mlrun/model_monitoring/db/tsdb/preaggregate.py

```diff
@@ -0,0 +1,221 @@
+                f"Available functions: {self._pre_aggregate_config.agg_functions}"
+            )
+
+    def can_use_pre_aggregates(
+        self, interval: Optional[str] = None, agg_funcs: Optional[list[str]] = None
+    ) -> bool:
+        """Check if pre-aggregates can be used for the given parameters."""
+        if not self._pre_aggregate_config or not interval:
+            return False
+
+        if interval not in self._pre_aggregate_config.aggregate_intervals:
+            return False
+
+        if agg_funcs:
+            return all(
+                func in self._pre_aggregate_config.agg_functions for func in agg_funcs
+            )
+
+        return True
+
+    def align_time_to_interval(
+        self, dt: datetime, interval: str, align_start: bool = True
+    ) -> datetime:
+        """Align datetime to interval boundaries."""
+        if not interval:
+            return dt
+
+        # Parse interval (e.g., "1h", "10m", "1d")
+        match = re.match(r"(\d+)([mhd]|M)", interval)
+        if not match:
+            return dt
+
+        amount, unit = int(match.group(1)), match.group(2)
+
+        if unit == "m":  # minutes
+            if align_start:
+                return dt.replace(second=0, microsecond=0) - timedelta(
+                    minutes=dt.minute % amount
+                )
+            else:
+                aligned_start = dt.replace(second=0, microsecond=0) - timedelta(
+                    minutes=dt.minute % amount
+                )
+                return aligned_start + timedelta(minutes=amount)
+        elif unit == "h":  # hours
+            if align_start:
+                return dt.replace(minute=0, second=0, microsecond=0) - timedelta(
+                    hours=dt.hour % amount
+                )
+            else:
+                aligned_start = dt.replace(
+                    minute=0, second=0, microsecond=0
+                ) - timedelta(hours=dt.hour % amount)
+                return aligned_start + timedelta(hours=amount)
+        elif unit == "d":  # days
+            if align_start:
+                return dt.replace(hour=0, minute=0, second=0, microsecond=0)
+            else:
+                return dt.replace(
+                    hour=0, minute=0, second=0, microsecond=0
+                ) + timedelta(days=amount)
+        elif unit == "M":  # months (approximate)
+            if align_start:
+                return dt.replace(day=1, hour=0, minute=0, second=0, microsecond=0)
+            else:
+                # Approximate month boundary
+                next_month = dt.replace(
+                    day=1, hour=0, minute=0, second=0, microsecond=0
+                )
+                if dt.month == 12:
+                    next_month = next_month.replace(year=dt.year + 1, month=1)
+                else:
+                    next_month = next_month.replace(month=dt.month + 1)
+                return next_month
+
+        return dt
+
+    def align_time_range(
```

Comment:
Need to take timezone into account as it could completely mess up the alignment.